### PR TITLE
Add aggregation-class with limit and orderby function

### PR DESCRIPTION
**Pull request by marcgeurts**
## Purpose
Add aggregation-class with limit and orderby function

### CODE REVIEW

File: docs/roadmap.md [hunk 1]
```diff
@@ -4,198 +4,198 @@
 
 ## current and next versions
 
-| method | Version 1.0 | Version 2.0 |
+| method | Version 1.0 | Version 1.1 | Version x |
 | --- | :---: | :---: |
```
Comment (tbolier) [inline docs/roadmap.md]: You miss an additional field here, when you view the roadmap via Github you will see the table is broken

File: src/Query/Aggregation/AbstractAggregation.php [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?php
+declare(strict_types = 1);
+
+namespace TBolier\RethinkQL\Query\Aggregation;
+
+use TBolier\RethinkQL\Query\AbstractQuery;
+
+abstract class AbstractAggregation extends AbstractQuery implements AggregationInterface
```
Comment (tbolier) [inline src/Query/Aggregation/AbstractAggregation.php]: We do not need an abstract if it is empty

File: src/Query/Aggregation/AggregationInterface.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+declare(strict_types = 1);
+
+namespace TBolier\RethinkQL\Query\Aggregation;
+
+use TBolier\RethinkQL\Response\ResponseInterface;
+
+interface AggregationInterface
```
Comment (tbolier) [inline src/Query/Aggregation/AggregationInterface.php]: This interface is missing the limit and orderBy method?

File: src/Query/Operation/Get.php [hunk 4]
```diff
@@ -4,11 +4,12 @@
 namespace TBolier\RethinkQL\Query\Operation;
 
 use TBolier\RethinkQL\Message\MessageInterface;
+use TBolier\RethinkQL\Query\AbstractQuery;
 use TBolier\RethinkQL\Query\QueryInterface;
 use TBolier\RethinkQL\RethinkInterface;
 use TBolier\RethinkQL\Types\Term\TermType;
 
-class Get extends AbstractOperation
+class Get extends AbstractQuery
```
Comment (tbolier) [inline src/Query/Operation/Get.php]: Get needs to support other operations like `Update` and `Update`, etc. so it needs to extend AbstractQuery

File: src/Query/Operation/TableCreate.php [hunk 5]
```diff
@@ -4,10 +4,11 @@
 namespace TBolier\RethinkQL\Query\Operation;
 
 use TBolier\RethinkQL\Message\MessageInterface;
+use TBolier\RethinkQL\Query\AbstractQuery;
 use TBolier\RethinkQL\RethinkInterface;
 use TBolier\RethinkQL\Types\Term\TermType;
 
-class TableCreate extends AbstractOperation
+class TableCreate extends AbstractQuery
```
Comment (tbolier) [inline src/Query/Operation/TableCreate.php]: TableCreate needs to support other operations like Update and Update, etc. so it needs to extend AbstractQuery. See the rethink.local data explorer autocomplete or Java API.

File: src/Query/Ordening.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
+<?php
+declare(strict_types = 1);
+
+namespace TBolier\RethinkQL\Query;
+
+use TBolier\RethinkQL\Message\MessageInterface;
+use TBolier\RethinkQL\RethinkInterface;
+use TBolier\RethinkQL\Types\Term\TermType;
+
+class Ordening extends AbstractQuery implements OrdeningInterface
```
Comment (tbolier) [inline src/Query/Ordening.php]: Is any operation possible after ordering? Should it really extend AbstractQuery?

File: src/Query/Table.php [hunk 7]
```diff
@@ -40,11 +42,27 @@ public function __construct(string $name, RethinkInterface $rethink, MessageInte
     /**
      * @inheritdoc
      */
-    public function get($value): OperationInterface
+    public function get($value): QueryInterface
```
Comment (tbolier) [inline src/Query/Table.php]: This is incorrect according to the ReQL API, see earlier comment.

File: src/Query/TableInterface.php [hunk 8]
```diff
@@ -3,13 +3,26 @@
...
 
 interface TableInterface extends OperationInterface
 {
     /**
      * @param string|int $value
-     * @return OperationInterface
+     * @return QueryInterface
      */
-    public function get($value): OperationInterface;
+    public function get($value): QueryInterface;
```
Comment (tbolier) [inline src/Query/TableInterface.php]: This is incorrect according to the ReQL API, see earlier comment.

File: src/Query/Ordening.php (hunk 6 above)
Comment (marcgeurts) [reply src/Query/Ordening.php]: Yes, more operations are possible.

File: docs/roadmap.md (hunk 1 above)
Comment (marcgeurts) [reply docs/roadmap.md]: fixed

File: src/Query/Aggregation/AbstractAggregation.php (hunk 2 above)
Comment (marcgeurts) [reply src/Query/Aggregation/AbstractAggregation.php]: It's not empty anymore

File: src/Query/Aggregation/AggregationInterface.php (hunk 3 above)
Comment (marcgeurts) [reply src/Query/Aggregation/AggregationInterface.php]: Fixed

File: src/Query/Operation/Get.php (hunk 4 above)
Comment (marcgeurts) [reply src/Query/Operation/Get.php]: Fixed

File: src/Query/Operation/TableCreate.php (hunk 5 above)
Comment (marcgeurts) [reply src/Query/Operation/TableCreate.php]: Fixed

**Pull request merged**